### PR TITLE
Add gitignore.

### LINKED ISSUE #4: Add a .gitignore
A .gitignore file is a magic text file that goes in the main directory of a git repository and tells git what files it doesn't need to track. In particular, local settings, build outputs (class/jar files), and autogenerated files (.DS_Store).

For more info, take a look at https://help.github.com/articles/ignoring-files/ and https://github.com/Open-RIO/.gitignore

Remember: don't add any line of code that you don't understand! Ask me if you have any questions :)

### PR DESCRIPTION
**Pull request by mlucas**
This resolves https://github.com/Tino-FRC-2473/FRC2018-Offseason/issues/4.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+###
+
+# Compiled class file
+*.class
+
+# Log file
+*.log
+
+# BlueJ files
+*.ctxt
```
Comment (tnarayanan) [inline .gitignore]: Not relevant because we are not using BlueJ

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
```
Comment (tnarayanan) [inline .gitignore]: We are not using Mobile Tools for Java, so this is not necessary

**Review by tnarayanan — CHANGES_REQUESTED** [review]
Some of the file extensions will not be used at all, so they do not need to be in the gitignore.

**Review by mvoodarla — APPROVED** [review]
I removed a few things that I thought were unnecessary so I think it's good to commit. We now need to remove all the files in the gitignore that have already been committed.

**Review by tnarayanan — APPROVED** [review]

**Pull request merged**